File: server/src/main/java/com/uten/imp/features/webinquiry/WebsiteInquiry.java
package com.uten.imp.features.webinquiry;

import com.uten.imp.common.domain.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

/**
 * 官网询盘（客户留言统一收件箱，综合营销）。
 *
 * <p>官网（Next.js 站点）询盘落库后推送到 {@code POST /api/website-inquiries/ingest}，
 * {@link #sourceId} 为官网 inquiries.id，重推幂等。官网后台保留只读副本，IMP 为处理主入口。
 *
 * <p>状态机：new → following → converted / closed；converted 必须已关联 {@link #clientId}。
 */
@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "website_inquiries")
public class WebsiteInquiry extends BaseEntity {

    /** 官网 inquiries.id（重推幂等键，唯一约束兜底）。 */
    @Column(name = "source_id", nullable = false, unique = true)
    private String sourceId;

    @Column(nullable = false)
    private String name;
    private String phone;
    private String email;
    private String company;

    /** 目标市场/国家。 */
    private String market;

    /** distributor/project/oem/designer/other */
    @Column(name = "customer_type")
    private String customerType;

    /** 目标市场要求的标准（如安装体系/认证要求，自由文本）。 */
    @Column(name = "required_standard")
    private String requiredStandard;

    @Column(name = "product_interest")
    private String productInterest;

    /** quotation/sample/technical/partnership/other */
    @Column(name = "request_type")
    private String requestType;

    @Column(name = "estimated_quantity")
    private String estimatedQuantity;

    @Column(name = "target_schedule")
    private String targetSchedule;

    @Column(name = "preferred_contact")
    private String preferredContact;

    @Column(nullable = false)
    private String message;

    /** contact/join/partner/product/studio */
    @Column(nullable = false)
    private String source = "contact";

    @Column(nullable = false)
    private String locale = "zh";

    /** new/following/converted/closed */
    @Column(nullable = false)
    private String status = "new";

    /** 跟进人（employees.id）。 */
    @Column(name = "assignee_employee_id")
    private UUID assigneeEmployeeId;

    /** 转客户后关联的客户主档 ID；跨 feature 查询经应用 Port 完成。 */
    @Column(name = "client_id")
    private UUID clientId;

    /** 最近一次跟进备注。 */
    private String note;

    @Column(name = "received_at", nullable = false)
    private Instant receivedAt = Instant.now();
}

File: lib/features/webinquiry/models/website_inquiry.dart
// 官网询盘模型
// 后端：server .../features/webinquiry（/api/website-inquiries）

import 'package:flutter/material.dart';

/// 官网询盘处理状态（与服务端状态机一致）。
enum WebsiteInquiryStatus {
  newOne('未处理', 0xFF3B82F6),
  following('跟进中', 0xFFF59E0B),
  converted('已转客户', 0xFF22C55E),
  closed('已关闭', 0xFF64748B);

  const WebsiteInquiryStatus(this.label, this.colorHex);
  final String label;
  final int colorHex;
  Color get color => Color(colorHex);

  static WebsiteInquiryStatus fromName(String? name) => switch (name) {
    'new' => WebsiteInquiryStatus.newOne,
    'following' => WebsiteInquiryStatus.following,
    'converted' => WebsiteInquiryStatus.converted,
    'closed' => WebsiteInquiryStatus.closed,
    _ => throw FormatException('Unknown website inquiry status: $name'),
  };

  String get wireName => switch (this) {
    WebsiteInquiryStatus.newOne => 'new',
    WebsiteInquiryStatus.following => 'following',
    WebsiteInquiryStatus.converted => 'converted',
    WebsiteInquiryStatus.closed => 'closed',
  };
}

/// 官网询盘（客户留言统一收件箱，综合营销处理）。
class WebsiteInquiry {
  const WebsiteInquiry({
    required this.id,
    required this.name,
    required this.message,
    required this.status,
    required this.receivedAt,
    this.company,
    this.phone,
    this.email,
    this.market,
    this.customerType,
    this.requiredStandard,
    this.productInterest,
    this.requestType,
    this.estimatedQuantity,
    this.targetSchedule,
    this.preferredContact,
    this.source = 'contact',
    this.locale = 'zh',
    this.assigneeName,
    this.clientId,
    this.clientName,
    this.note,
  });

  final String id;
  final String name;
  final String message;
  final WebsiteInquiryStatus status;
  final DateTime receivedAt;

  final String? company;
  final String? phone;
  final String? email;
  final String? market;
  final String? customerType;
  final String? requiredStandard;
  final String? productInterest;
  final String? requestType;
  final String? estimatedQuantity;
  final String? targetSchedule;
  final String? preferredContact;
  final String source;
  final String locale;

  /// 跟进人姓名快照（服务端按 assigneeEmployeeId 解析）。
  final String? assigneeName;

  /// 转客户后关联的客户主档。
  final String? clientId;
  final String? clientName;

  /// 最近一次跟进备注。
  final String? note;

  /// 列表卡片用的一句话摘要。
  String get subtitle => [
    company,
    market,
  ].whereType<String>().where((s) => s.isNotEmpty).join(' · ');
}
